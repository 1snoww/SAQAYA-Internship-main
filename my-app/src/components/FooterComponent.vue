<template>
  <footer class="footer">
    <p>Â© 2025 SAQAYA. All rights reserved.</p>
    <p>Built with love by the Internship Team</p>
  </footer>
</template>

<style scoped>
.footer {
  background-color: #2c3e50;
  color: white;
  text-align: center;
  padding: 20px;
  font-size: 0.9rem;
  margin-top: 40px;
}
</style>
