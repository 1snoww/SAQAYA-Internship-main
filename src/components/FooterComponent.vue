<template>
  <footer class="footer">
    <div class="container">
      <p>Â© {{ year }} SAQAYA Shop. All rights reserved.</p>
    </div>
  </footer>
</template>

<script setup lang="ts">
import { ref } from 'vue'

const year = ref(new Date().getFullYear())
</script>

<style scoped>
.footer {
  background-color: #1f2e3d;
  color: white;
  padding: 20px 0;
  text-align: center;
}

.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 20px;
}
</style>
