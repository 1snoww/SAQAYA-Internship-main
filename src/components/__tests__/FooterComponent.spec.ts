import { mount } from '@vue/test-utils'
import FooterComponent from '../FooterComponent.vue'
import { describe, it, expect } from 'vitest'

describe('FooterComponent.vue', () => {
  it('renders © symbol', () => {
    const wrapper = mount(FooterComponent)
    expect(wrapper.text().toLowerCase()).toContain('©')
  })

  it('matches snapshot', () => {
    const wrapper = mount(FooterComponent)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
