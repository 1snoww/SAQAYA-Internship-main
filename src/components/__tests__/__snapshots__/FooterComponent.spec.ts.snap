// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterComponent.vue > matches snapshot 1`] = `
"<footer data-v-7490b180="" class="footer">
  <div data-v-7490b180="" class="container">
    <p data-v-7490b180="">Â© 2025 SAQAYA Shop. All rights reserved.</p>
  </div>
</footer>"
`;
